File: elements_of_programming_interviews/src/C6_Arrays/Q0_BootCamp/Python/EvenOddArray.py
import sys
import random


def even_odd(a):
    next_even, next_odd = 0, len(a) - 1
    while next_even < next_odd:
        if a[next_even] % 2 == 0:
            next_even += 1
        else:
            temp = a[next_even]
            a[next_even] = a[next_odd]
            a[next_odd] = temp;
            next_odd -= 1


def main():
    for _ in range(10000):
        n = int(sys.argv[1]) if len(sys.argv) == 2 else random.randint(1, 20)
        a = [random.randint(0, 20) for _ in range(n)]
        even_odd(a)


if __name__ == '__main__':
    main()

File: src/ElementsOfProgrammingInterviews/src/C6_Arrays/Q0_BootCamp/Java/EvenOddArray.java
package ElementsOfProgrammingInterviews.src.C6_Arrays.Q0_BootCamp.Java;

import java.util.Random;

public class EvenOddArray {
    public static void evenOdd(int[] A) {
        int nextEven = 0, nextOdd = A.length - 1;
        while (nextEven < nextOdd) {
            if (A[nextEven] % 2 == 0) {
                nextEven++;
            } else {
                int temp = A[nextEven];
                A[nextEven] = A[nextOdd];
                A[nextOdd--] = temp;
            }
        }
    }

    public static void main(String[] args) {
        Random rnd = new Random();
        int[] A = new int[0];
        for (int i = 0; i < 1000; i++) {
            A = new int[rnd.nextInt(20) + 1];
            for (int j = 0; j < A.length; j++) {
                A[j] = rnd.nextInt(20);
            }
        }
        evenOdd(A);
    }
}
